### PR TITLE
adding sql schema, and register endpoint

### PR DESCRIPTION
**Pull request by SamTheBosssss**
Note the register endpoint is used for internal testing.

### CODE REVIEW

File: flask-api/flaskr/auth.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+@bp.route('/register', methods=('POST',))
+def register():
+    try:
+        username = request.json['username']
+        password = request.json['password']
+        full_name = request.json['full_name']
+        user_type = request.json['user_type']
+    except:
+        abort(400, "bad inputs")
```
Comment (benedict-soh) [inline flask-api/flaskr/auth.py]: - [x] Change to `abort(400, "Bad Request")`

File: flask-api/flaskr/auth.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+    db = get_db()
+
+    # check for duplicate username 
+    if db.execute(
+        'SELECT user_id FROM user_cred WHERE username = ?', (username,)
+    ).fetchone() is not None:
+        abort(400, 'User {} is already registered.'.format(username))
```
Comment (benedict-soh) [inline flask-api/flaskr/auth.py]: - [x] Change to `abort(400, "Bad Request")`

File: flask-api/flaskr/auth.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        full_name = request.json['full_name']
+        user_type = request.json['user_type']
+    except:
+        abort(400, "bad inputs")
+
+    db = get_db()
+
+    # check for duplicate username 
+    if db.execute(
+        'SELECT user_id FROM user_cred WHERE username = ?', (username,)
```
Comment (benedict-soh) [inline flask-api/flaskr/auth.py:26]: Is the comma in `(username,)` a typo?

**Review by benedict-soh — CHANGES_REQUESTED** [review]
Pls address comments

File: flask-api/flaskr/auth.py (hunk 3 above)
Comment (SamTheBosssss) [reply flask-api/flaskr/auth.py:26]: nope, needs to be a list like object

**Review by benedict-soh — APPROVED** [review]

**Pull request merged**